### PR TITLE
SQL: Allow skip of bwc tests on `check` task

### PR DESCRIPTION
**Pull request by matriv**
Bwc tests can consume much time to build and to run so it's nice to be
able to skip them when running the `check` task on the SQL module.

Introduce a project property `skipBwc` so one can use:
```
./gradlew -p x-pack/plugin/sql check -PskipBwc
```
to skip them.

### CODE REVIEW

**Review by costin — APPROVED** [review]
LGTM

**Review by astefan — APPROVED** [review]
LGTM

**Comment by rjernst** [comment]
I don't think we should be added such properties. Bwc tests on check are already limited to snapshot versions, so from master, this is only 7.x that is tested against. If you want to run a specific test of tests, for example unit tests or rest tests, then use the appropriate tasks directly. Skipping creation of tasks is not what we want. /cc @mark-vieira

**Comment by mark-vieira** [comment]
I agree with Ryan. These magic flags are not very discoverable. I'd prefer if you want some kind of "shortcut" for the sql project you should add a specific task for this that depends on all tests except for the BWC ones.

File: x-pack/plugin/sql/build.gradle [hunk 1]
```diff
@@ -132,3 +132,11 @@ tasks.register("regen") {
     }
   }
 }
+
+task checkNoBwc {
+  dependsOn 'check'
```
Comment (mark-vieira) [inline x-pack/plugin/sql/build.gradle:137]: This wasn't exactly what I was thinking. What I was saying is that `checkNoBwc` should depend on `test`, `integTest`, `yamlRestTest`, etc. That way it runs those w/o running all the BWC stuff.

**Review by mark-vieira — CHANGES_REQUESTED** [review]

**Review by mark-vieira — APPROVED** [review]
LGTM

**Pull request merged**